### ISSUE TITLE
How to use UNI for segmentation task?

### ISSUE THREAD

**Issue by Leonngm** [mahmoodlab/UNI#12]
Dear authors,
thanks for your great work!
In your publication you mention that you also used UNI for segmentation of images. Could you please provide insights how to use UNI for segmentation tasks?
Thank you!
Kind regards.

**Comment by lely475** [comment]
>  would it be possible to reopen the issue and give a more detailed answer to the question?

I would second this. 

From the paper it appears that UNI was used as a backbone with ViT Adapter (for including hierarchical feaures) and Mask2Former head for segmentation using https://github.com/czczup/ViT-Adapter. However, I don't find this code bases very straightforward to use, so would appreciate more insights in how to include UNI as backbone in the ViT-Adapter infrastructure.

This is the section from the paper on **ROI-level cell type segmentation**:

> For training and evaluation of ROI-level cell type segmentation tasks, we follow previous works in using Mask2Former, which is a flexible framework commonly used for evaluating off-the-shelf performance of pretrained vision encoders In the case of the ViT architecture, which is nonhierarchical, we additionally use the ViT-Adapter framework alongside the Mask2Former head. For both ViT-Adapter and Mask2Former, we use the same hyper-parameters used for ADE20k semantic segmentation. Specifically, we use the AdamW optimizer along with a step learning rate schedule. The initial learning rate was set to 0.0001 and a weight decay of 0.05 was applied. To adjust the learning rate specifically for the backbone, we apply a learning rate multiplier of 0.1. Additionally, we decay the learning rate by a factor of 10 at 0.9 and 0.95 fractions of the total number of training steps. For all backbones, we fine-tune the full model for 50 epochs with a batch size of 16. The model’s performance on the validation set is evaluated every 5 epochs, and the optimal model based on validation performance is saved for testing. To augment the data, we use the large-scale jittering (LSJ) augmentation, with a random scale sampled from a range of 0.5–2.0, followed by a fixed size crop to 896 × 896 pixels to accommodate the size constraints of CTransPath. At inference time, we resize the image dimensions to their nearest multiples of 224.

**Comment by lsnls** [comment]
I would second this, too.
